connect/api/src/main/java/org/apache/kafka/connect/data/ConnectSchema.java

```diff
@@ -230,7 +237,9 @@ public static void validateValue(Schema schema, Object value) {
             }
         }
         if (!foundMatch)
-            throw new DataException("Invalid Java object for schema type " + schema.type() + ": " + value.getClass());
+            throw new DataException("Invalid Java object for schema type " + schema.type()
```

Comment:
I have removed "required" from error message. we can judge if it's required field from `schema.isOptional()` , but we may don't care about it when the value is not null.